src/benchmark_runner.cc

```diff
@@ -228,20 +232,21 @@ BenchmarkRunner::BenchmarkRunner(
     : b(b_),
       reports_for_family(reports_for_family_),
       parsed_benchtime_flag(ParseBenchMinTime(FLAGS_benchmark_min_time)),
-      min_time(ComputeMinTime(b_, parsed_benchtime_flag)),
-      min_warmup_time((!IsZero(b.min_time()) && b.min_warmup_time() > 0.0)
-                          ? b.min_warmup_time()
-                          : FLAGS_benchmark_min_warmup_time),
+      min_time(FLAGS_benchmark_dry_run ? 0 : ComputeMinTime(b_, parsed_benchtime_flag)),
```

Comment:
this is starting to get complex.  i wonder if it's worth moving these into the body of the constructor and having 

```
if (FLAGS_benchmark_dry_run) {
  min_time = 0;
  min_warmup_time = 0;
  ...
} else {
  min_time = ComputeMinTime(b_, parsed_benchtime_flag);
  ...
}
```

wdyt?

Reply:
I agree, I think for the long run it will help reduce complexity. i'll revise that.

Reply:
We could also revert to the original logic and then add a simple if statement for the dry run case.

```
   ...
  iters(has_explicit_iteration_count
                ? ComputeIters(b_, parsed_benchtime_flag)
                : 1),
      perf_counters_measurement_ptr(pcm_) {
  if (FLAGS_benchmark_dry_run) {
    min_time = 0;
    min_warmup_time = 0;
    warmup_done = true;
    repeats = 1;
    iters = 1;
  }
  ...
  ```

Reply:
yes perhaps.  i'm not a fan of this being "hidden" but it does ensure the fields are initialized correctly on construction.